android-components/components/concept/storage/src/main/java/mozilla/components/concept/storage/HistoryStorage.kt

```diff
@@ -225,6 +223,9 @@ enum class VisitType(val type: Int) {
     DOWNLOAD(7),
     FRAMED_LINK(8),
     RELOAD(9),
+
+    // Internal visit type used for meta data updates. Doesn't represent an actual page visit.
+    UPDATE_PLACE(10),
```

Comment:
Couple questions for my own edification:
I see a kdoc for this type that says these match Firefox desktop. Is that still true with this change?

More broadly, do you happen to know if there's reasoning behind redeclaring these enums on our side? Seems a little pointless at first glance to me to not just use the UniFFI types everywhere

Reply:
Great Question! So `update_place` is the only departure from desktop: https://searchfox.org/mozilla-central/source/toolkit/components/places/nsINavHistoryService.idl#947 and is specifically used only in android, so I'd rather not shake the boat too much without understanding the historical reasons here.

I'd ideally like to use only the Uniffi'd version and remove all this extra code but it would break some practices in android-components.

1. In the "abstract" History storage we have a VisitType defined here: https://github.com/mozilla-mobile/firefox-android/blob/471ef6ed0f7371996e72a3c432284d5a0bc12f99/android-components/components/concept/storage/src/main/java/mozilla/components/concept/storage/HistoryStorage.kt#L211-L228
 - This is what is used across the app, however this defined type is in the `interface HistoryStorage` https://github.com/mozilla-mobile/firefox-android/blob/471ef6ed0f7371996e72a3c432284d5a0bc12f99/android-components/components/concept/storage/src/main/java/mozilla/components/concept/storage/HistoryStorage.kt#L10

which means I'd have to remove this interface type and explicitly use the Syncs storage which would break the current "swappable storage layer" paradigm that currently exists. The closest thing I could do is just simply keeping the conversion types. Totally open to ideas to get rid of those! I think overall it still shaves a couple of abstraction layers tho.

Reply:
Ah, the swappable implementation thing makes sense in theory. Let's not rock the boat to hard in that case and just keep the patch as is